main.cpp

```diff
@@ -77,25 +77,21 @@ struct CompareDaFunk                                //4
 struct U
 {
     float oldFunk { 0 }, newFunk { 0 };
-    float regularFuncA(U* that, float* updatedFunk )      //12
+    float regularFuncA(float* updatedFunk )      //12
     {
-        std::cout << "U's oldFunk value: " << that->oldFunk << std::endl;
-        
-        // check before deref'ing that funk
-        if (updatedFunk == nullptr)
-            return 0;
+        std::cout << "U's oldFunk value: " << oldFunk << std::endl;
 
-        that->oldFunk = *updatedFunk;
-        std::cout << "U's oldFunk updated value: " << that->oldFunk << std::endl;
-        while( std::abs(that->newFunk - that->oldFunk) > 0.001f )
+        oldFunk = *updatedFunk;
```

Comment:
FIXME: you're still using `updatedFunk` without checking if it's a nullptr first!
- [ ] revise